lib/gen_rmq/consumer/queue_configuration.ex

```diff
@@ -56,9 +56,11 @@ defmodule GenRMQ.Consumer.QueueConfiguration do
     # This is done in order to be backwards compatible.
     # If one day those two keywords are removed from the
     # init, then this function can be removed as well.
+    queue_option_arguments = Keyword.get(option_list, :arguments, [])
+
     word_list
-    |> remove_keyword(option_list[:arguments], %{name: "x-expires", word: :ttl})
-    |> remove_keyword(option_list[:arguments], %{name: "x-max-priority", word: :max_priority})
+    |> remove_keyword(queue_option_arguments, %{name: "x-expires", word: :ttl})
+    |> remove_keyword(queue_option_arguments, %{name: "x-max-priority", word: :max_priority})
```

Comment:
Maybe also make remove_keyword function into multi-clause function where empty option_arguments is matched and word_list is returned.
```
defp remove_keyword(word_list, [], _argument), do: word_list

defp remove_keyword(word_list, option_arguments, argument) do
    case Enum.find(option_arguments, fn arg -> elem(arg, 0) == argument.name end) do
      nil -> word_list
      _ -> Keyword.delete(word_list, argument.word)
    end
  end
```
It would make the intent clearer. In cases where option_arguments is empty, we want to return word_list.

Reply:
@Shemeikka good idea, will do 👍